stackit/internal/services/serverbackup/schedule/resource.go

```diff
@@ -0,0 +1,539 @@
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("backup_schedule_id"), intId)...)
+	tflog.Info(ctx, "Serverbackup schedule state imported")
+}
+
+func mapResourceFields(ctx context.Context, schedule *serverbackup.BackupSchedule, model *scheduleResourceModel) error {
+	if schedule == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+	if schedule.Id == nil {
+		return fmt.Errorf("response id is nil")
+	}
+
+	model.BackupScheduleId = types.Int64PointerValue(schedule.Id)
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		model.ServerId.ValueString(),
+		strconv.FormatInt(model.BackupScheduleId.ValueInt64(), 10),
+	}
+	model.ID = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+	model.Name = types.StringPointerValue(schedule.Name)
+	model.Rrule = types.StringPointerValue(schedule.Rrule)
+	model.Enabled = types.BoolPointerValue(schedule.Enabled)
+	if schedule.BackupProperties == nil {
+		model.BackupProperties = nil
+		return nil
+	}
+	ids, diags := types.ListValueFrom(ctx, types.StringType, schedule.BackupProperties.VolumeIds)
+	if diags.HasError() {
+		return fmt.Errorf("failed to map hosts: %w", core.DiagsToError(diags))
+	}
+	model.BackupProperties = &scheduleBackupPropertiesModel{
+		BackupName:      types.StringValue(*schedule.BackupProperties.Name),
+		RetentionPeriod: types.Int64Value(*schedule.BackupProperties.RetentionPeriod),
+		VolumeIds:       ids,
+	}
+	return nil
+}
+
+// If already enabled, just continues
+func enableBackupsService(ctx context.Context, model *scheduleResourceModel, client *serverbackup.APIClient) error {
+	projectId := model.ProjectId.ValueString()
+	serverId := model.ServerId.ValueString()
+	enableServicePayload := serverbackup.EnableServicePayload{}
+
+	tflog.Debug(ctx, "Enabling server backup service")
+	err := client.EnableService(ctx, projectId, serverId).EnableServicePayload(enableServicePayload).Execute()
+	if err != nil {
+		if strings.Contains(err.Error(), "Tried to activate already active service") {
+			tflog.Debug(ctx, "Service for server backup already enabled")
+			return nil
+		}
+		return fmt.Errorf("Failed to enable server backup service: %w", err)
+	}
+	tflog.Info(ctx, "Enabled server backup service")
+	return nil
+}
+
+// Disables only if no backup schedules are present and no backups are present
+func disableBackupsService(ctx context.Context, model *scheduleResourceModel, client *serverbackup.APIClient) error {
+	tflog.Debug(ctx, "Disabling server backup service (in case there are no backups and no backup schedules)")
+
+	projectId := model.ProjectId.ValueString()
+	serverId := model.ServerId.ValueString()
+
+	tflog.Debug(ctx, "Checking for existing backup schedules")
+	schedules, err := client.ListBackupSchedules(ctx, projectId, serverId).Execute()
+	if err != nil {
+		return fmt.Errorf("Failed to disable server backup service: %w", err)
```

Comment:
E.g. this error message could be simply `list existing backup schedules`, because the context of the caller is already indicating that we are trying to disable the service